### PR TITLE
 add `onSearchChange(direction: 'left'|'right') => void ` of Transfer

### LINKED ISSUE #4444: Support `Transfer[onSearchChange]`
`Transfer[onSearchChange]` will be triggered while user input in search box.

### PR DESCRIPTION
**Pull request by RaoHai**
- close  #4444

### CODE REVIEW

File: components/transfer/index.zh-CN.md [hunk 1]
```diff
@@ -35,6 +35,7 @@ title: Transfer
 | notFoundContent | 当列表为空时显示的内容 | React.ReactNode | '列表为空'  |
 | footer | 底部渲染函数 | (props): React.ReactNode |  |
 | lazy | Transfer 使用了 [react-lazy-load](https://github.com/loktar00/react-lazy-load) 优化性能，这里可以设置相关参数 | Object | `{ height: 32, offset: 32 }` |
+| onSearchChange | 搜索框内容时改变时的回调函数 | (direction: 'left' | 'right', evenet: Event): void | - |
```
Comment (afc163) [inline components/transfer/index.zh-CN.md]: event

File: components/transfer/index.en-US.md [hunk 2]
```diff
@@ -33,6 +33,8 @@ One or more elements can be selected from either column, one click on the proper
 | notFoundContent | Text to display when a column is empty. | React.ReactNode | 'The list is empty'  |
 | footer | A function used for rendering the footer. | (props): React.ReactNode |  |
 | lazy | property of [react-lazy-load](https://github.com/loktar00/react-lazy-load) for lazy rendering items | Object | `{ height: 32, offset: 32 }` |
+| onSearchChange | A callback function which is executed when search field are changed | (direction: 'left' | 'right', evenet: Event): void | - |
```
Comment (benjycui) [inline components/transfer/index.en-US.md]: 这儿也拼错了。。

Comment (RaoHai) [inline components/transfer/index.en-US.md]: 手滑。。

**Comment by benjycui** [comment]
😢  it is a feature.

Please PR to `feature-2.7`

**Comment by RaoHai** [comment]
@benjycui #4464

**Pull request closed without merging**